include/tvm/runtime/packed_func.h

```diff
@@ -904,8 +905,18 @@ inline const char* TypeCode2Str(int type_code) {
     case kFuncHandle: return "FunctionHandle";
     case kModuleHandle: return "ModuleHandle";
     case kNDArrayContainer: return "NDArrayContainer";
-    default: LOG(FATAL) << "unknown type_code="
-                        << static_cast<int>(type_code); return "";
+    default:
+      // This is (potentially) the case of a custom type code.
+      // TODO(gus): should only a range of codes be allowed for customs?
+
+      if (GetTypeRegistered(type_code)) {
```

Comment:
Here, we should

1. Use a simpler test: declare a type code max somewhere and just compare against that.
2. Declare a  `TVM_DLL` function here for getting a custom datatype's name; define it in the non-public implementation.